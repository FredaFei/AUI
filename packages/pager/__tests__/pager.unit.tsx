import * as renderer from 'react-test-renderer'
import React from 'react'
import {mount} from 'enzyme'

import Pager from '../pager'

describe('测试 Pager 以下行为', () => {
  it('Pager 存在', () => {
    const json = renderer.create(<Pager defaultCurrent={4} total={10}/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Pager 设置 defaultCurrent & total', () => {
    const wrapper = mount(<Pager defaultCurrent={4} total={5}/>)
    expect(wrapper.props().defaultCurrent).toEqual(4)
    expect(wrapper.props().total).toEqual(5)
  })
  it('Pager 设置 快捷跳转', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Pager onChange={onChange} visibleQuickJumper total={10}/>)
    const input = wrapper.find('.am-pager-input-page').at(0);
    input.simulate('keypress', {keyCode: 13, which: 13,target: { value: 12}})
    expect(onChange).toHaveBeenCalled();
    input.simulate('keypress', {keyCode: 13, which: 13,target: { value: 3}})
    expect(onChange).toHaveBeenCalled();
  });

  it('Pager 手动触发 onChange', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Pager visibleQuickJumper current={3} total={10} onChange={onChange}/>)
    wrapper.find('.am-pager-list-item').at(1).simulate('click')
    expect(onChange).toHaveBeenCalledWith(2);
  })
  it('Pager 点击加载更多页', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Pager current={6} total={10} onChange={onChange}/>)
    wrapper.find('.am-pager-split').at(1).simulate('click')
    expect(onChange).toHaveBeenCalledWith(10);
    wrapper.setProps({ current: 6 });
    wrapper.find('.am-pager-split').at(0).simulate('click')
    expect(onChange).toHaveBeenCalledWith(1);

  })
  it('Pager 点击上一页', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Pager visibleQuickJumper current={3} total={10} onChange={onChange}/>)
    wrapper.find('.am-pager-prev').simulate('click')
    expect(onChange).toHaveBeenCalledWith(2);
  })
  it('Pager 点击下一页', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Pager visibleQuickJumper current={3} total={10} onChange={onChange}/>)
    wrapper.find('.am-pager-next').simulate('click')
    expect(onChange).toHaveBeenCalledWith(4);
  })
})