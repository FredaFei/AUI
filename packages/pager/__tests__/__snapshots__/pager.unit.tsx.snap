// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Pager 以下行为 Pager 存在 1`] = `
<div
  className="am-pager-wrapper "
>
  <span
    className="am-pager-prev am-pager-disabled"
    onClick={[Function]}
  >
    <svg
      className="am-icon "
      style={Object {}}
    >
      <use
        xlinkHref="#left"
      />
    </svg>
  </span>
  <ul
    className="am-pager-list"
  >
    <li
      className="am-pager-list-item am-pager-list-item-active"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="am-pager-list-item"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="am-pager-list-item"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="am-pager-list-item am-pager-split"
      onClick={[Function]}
    >
      <span>
        ...
      </span>
    </li>
    <li
      className="am-pager-list-item"
      onClick={[Function]}
    >
      10
    </li>
  </ul>
  <span
    className="am-pager-next"
    onClick={[Function]}
  >
    <svg
      className="am-icon "
      style={Object {}}
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </span>
</div>
`;
